src/main/java/jsg/Main.java

```diff
@@ -67,12 +74,23 @@ public static void main(String[] args) throws IOException {
 
     }
 
-    for (Entry<String, List<String>> entry : uniqueTraces.entrySet()) {
-      for (String header : entry.getValue()) {
-        System.out.println(header);
-      }
+    uniqueTraces.entrySet().stream()
+        .sorted(Map.Entry.comparingByKey(
+            Comparator.comparingInt(Main::toStateIdx)))
+        .map(header -> String.join("", header.getValue()) + header)
+        .forEach(System.out::println);
+  }
+
+  private static boolean isInState(String trace, String state) {
+    return trace.startsWith(PREFIX + state);
+  }
 
-      System.out.println(entry.getKey());
+  private static int toStateIdx(String trace) {
+    for (int index = 0; index < STATES.length; index++) {
```

Comment:
Another possible way to implement this would be the following.  This makes less parsing assumptions and does not assume what the states are.  However you have no control of the order.

```java
  private static int toStateIdx(String trace) {
    for(State state : State.values()) {
      if(trace.contains(" "+state.name())) {
        return state.ordinal();
      }
    }

    return State.values().length;
  }
```

Reply:
I think the trade off here is that, if some class name contains BLOCKED in it - it would lead to wrong result.

Reply:
That is a good point.   Could do `contains(PREFIX+state.name())`.